IPython/core/interactiveshell.py

```diff
@@ -2781,6 +2784,15 @@ def run_ast_nodes(self, nodelist, cell_name, interactivity='last_expr',
         if not nodelist:
             return
 
+        if interactivity == 'last_expr_or_assign':
+            if isinstance(nodelist[-1], ast.Assign):
+                asg = nodelist[-1]
+                if len(asg.targets) == 1:
+                    nnode = ast.Expr(ast.Name(asg.targets[0].id, ast.Load()))
```

Comment:
(That `%%ast` magic comes from https://github.com/llllllllll/codetransformer/blob/master/codetransformer/__init__.py#L23-L35, btw.  That might be relevant to your interests if you're working on ast stuff.)